runtime/compiler/control/JitDump.cpp

```diff
@@ -638,6 +638,13 @@ runJitdump(char *label, J9RASdumpContext *context, J9RASdumpAgent *agent)
          // get current compilation
          TR::Compilation *comp = threadCompInfo->getCompilation();
 
+         // Printing the crashed thread trees or any similar operation on the crashed thread may result in having to
+         // acquire VM access (ex. to get a class signature). Other VM events, such as VM shutdown or the GC unloading
+         // classes may cause compilations to be interrupted. Because the crashed thread is not a diagnostic thread,
+         // the call to print the crashed thread IL may get interrupted and the jitdump will be incomplete. We prevent
+         // this from occuring by disallowing interruptions until we are done generating the jitdump.
+         TR::CompilationInfoPerThreadBase::UninterruptibleOperation generateJitDumpForCrashedThread(*threadCompInfo);
```

Comment:
If the goal is to prevent interruption during `dumpCuurentIL` isn't better to move this line down, just above `dumpCurrentIL(comp, crashedThread, jitConfig, logFile);`

Reply:
No the goal is to guard both dumping the IL and the request to recompile. If I understood the path correctly it is possible for us to hit a yield point on our way to queue the JitDump compile. Is that correct?

Reply:
Even if it is not the case someone can still introduce such code in the future, unbeknownst to them that we don't want interruptions on our way to queue the JitDump compile. So it is best to guard the whole thing. Do you agree?

Reply:
I agree.